tests/main/views/test_briefs.py

```diff
@@ -1242,3 +1242,239 @@ def test_supplier_is_eligible_even_if_does_not_supply_in_outcome_location(self):
+            brief_response.award_details = {'pending': True}
+            db.session.add(brief_response)
+            db.session.commit()
+
+            res = self._post_to_award_endpoint({'brief_response_id': brief_response.id})
+            assert res.status_code == 200
+
+    def test_400_if_no_updated_by_in_payload(self):
+        res = self.client.post(
+            self.award_url,
+            data=json.dumps({"brief_response_id": 1}),
+            content_type="application/json"
+        )
+        assert res.status_code == 400
+        error = json.loads(res.get_data(as_text=True))['error']
+        assert "'updated_by' is a required property" in error
+
+    @pytest.mark.parametrize('status', ['draft', 'live', 'withdrawn', 'awarded'])
+    def test_400_if_awarding_a_brief_response_to_a_non_closed_brief(self, status):
+        self.setup_dummy_briefs(1, status=status)
+        res = self._post_to_award_endpoint({'brief_response_id': 1})
+        data = json.loads(res.get_data(as_text=True))
+        assert res.status_code == 400
+        assert data['error'] == "Brief is not closed"
+
+    def test_400_if_awarding_a_draft_brief_response_to_a_closed_brief(self):
+        with self.app.app_context():
+            self.setup_dummy_briefs(1, status="closed")
+            self.setup_dummy_suppliers(1)
+            brief_response = BriefResponse(brief_id=1, supplier_id=0, data={})
+            db.session.add(brief_response)
+            db.session.commit()
+
+            res = self._post_to_award_endpoint({'brief_response_id': brief_response.id})
+            data = json.loads(res.get_data(as_text=True))
+            assert res.status_code == 400
+            assert data['error'] == "BriefResponse cannot be awarded for this Brief"
+
+    def test_400_if_awarding_a_brief_response_that_does_not_relate_to_that_brief(self):
+        with self.app.app_context():
+            self.setup_dummy_briefs(2, status="closed")
+            self.setup_dummy_suppliers(1)
+            brief_response = BriefResponse(brief_id=2, supplier_id=0, submitted_at=datetime.utcnow(), data={})
+            db.session.add(brief_response)
+            db.session.commit()
+
+            # We've set up brief response for brief ID 2 but attempt to award it for brief 1
+            res = self._post_to_award_endpoint({'brief_response_id': brief_response.id})
+
+            data = json.loads(res.get_data(as_text=True))
+            assert res.status_code == 400
+            assert data['error'] == "BriefResponse cannot be awarded for this Brief"
+
+
+class TestBriefAwardDetails(FrameworkSetupAndTeardown):
+
+    award_url = "/briefs/1/award/{}/contract-details"
+    valid_payload = {
+        "award_details": {
+            "awardedContractStartDate": "2020-12-31",
+            "awardedContractValue": "99.95",
+        },
+        "updated_by": "user@email.com"
+    }
+
+    def _post_to_award_details_endpoint(self, payload, brief_response_id):
```

Comment:
Could create a base class for `_get_brief_response_audit_events` and `_post_to_award_details_endpoint`. Not a dealbreaker.